### PR TITLE
Add --all flag to dolt push

### PR DESCRIPTION
**Pull request by druvv**
In git, the `--all` flag can be specified which pushes all local branches. In addition, the `-u` flag can be specified which also tracks those branches.

This PR adds the `--all` flag to `dolt push`. It will be used in the migration jobs to push a migrated repo to an empty remote.

### CODE REVIEW

File: go/cmd/dolt/commands/push.go [hunk 1]
```diff
@@ -88,6 +89,53 @@ func (cmd PushCmd) Exec(ctx context.Context, commandStr string, args []string, d
...
+		verr := errhand.BuildDError("failed to fetch branches").AddCause(err).Build()
+		return HandleVErrAndExitCode(verr, usage)
+	}
+
+	if len(apr.Args) > 1 {
+		verr := errhand.BuildDError("only <remote> may be specified with --all").SetPrintUsage().Build()
+		return HandleVErrAndExitCode(verr, usage)
+	}
+
+	remoteName := "origin"
```
Comment (zachmu) [inline go/cmd/dolt/commands/push.go:111]: Is this what git does? Can't tell from the docs.

Normally `git push` without a remote named will push to the upstream tracking branch, but that can't be the case with `--all`

File: go/cmd/dolt/commands/push.go [hunk 2]
```diff
@@ -88,6 +89,53 @@ func (cmd PushCmd) Exec(ctx context.Context, commandStr string, args []string, d
...
 
+	if apr.Contains(cli.AllFlag) {
+		return cmd.pushAllBranches(ctx, apr, dEnv, usage)
+	}
+
+	return cmd.doPush(ctx, apr, dEnv, usage)
+}
+
+func (cmd PushCmd) pushAllBranches(ctx context.Context, apr *argparser.ArgParseResults, dEnv *env.DoltEnv, usage cli.UsagePrinter) int {
+	branches, err := dEnv.DoltDB.GetBranches(ctx)
```
Comment (zachmu) [inline go/cmd/dolt/commands/push.go:100]: This includes e.g. remote branches too, right? Or are those stored under a different ref namespace?

File: integration-tests/bats/remotes.bats [hunk 3]
```diff
@@ -2188,3 +2188,87 @@ SQL
...
+
+    # Push branch1 and branch2 to origin
+    run dolt push --all
+    [ $status -eq 0 ]
+
+    # Verify that branch1 and branch2 were pushed by checking them out in separate repo
+    cd ../repo2
+    dolt fetch
+    run dolt checkout branch1
+    [ $status -eq 0 ]
```
Comment (zachmu) [inline integration-tests/bats/remotes.bats:2230]: If your only check is `status eq 0`, you can just omit `run`, same thing

**Review by zachmu — APPROVED** [review]
LGTM, just a couple questions about weird edge cases
